edx_rbac/decorators.py

```diff
@@ -0,0 +1,35 @@
+"""
+Taken from https://github.com/escodebar/django-rest-framework-rules/blob/master/rest_framework_rules/decorators.py.
+"""
+from __future__ import absolute_import
+
+
+def permission_required(*permissions, **decorator_kwargs):
+    """
+    :param permissions: Permissions added via django_rules add_perm
+    :param decorator_kwargs: Arguments for permission checks
+    :return: decorator
+    """
+
+    def decorator(view):
+        def wrapped_view(self, request, *args, **kwargs):
+
+            fn = decorator_kwargs.get('fn', None)
+            if callable(fn):
+                obj = fn(request, *args, **kwargs)
+            else:
+                obj = fn
+
+            missing_permissions = [perm for perm in permissions
+                                   if not request.user.has_perm(perm, obj)]
```

Comment:
Would the `has_perm` be relying on the the access checks we have in utils.py? As in, would the bool returned by the `has_perm` call be taking into account the jwt/database access checks that are defined in the checks we have in utils.py?

Reply:
The rules predicate function e.g. https://github.com/edx/ecommerce/pull/2148/files#diff-222890e157cea486d8e170eafa8dca5eR20
can have customized checks including those utilizing JWT